pkg/evaluator/evaluator.go

```diff
@@ -35,7 +41,13 @@ func (e *Evaluator) Run(input string) {
 	}
 }
 
+var ErrStopped = newError("stopped")
+
 func (e *Evaluator) Eval(node parser.Node) Value {
+	if e.Stopped {
+		return ErrStopped
+	}
+	e.yield()
```

Comment:
Perhaps a comment here to say why this yield exists?

```
// Yield to give JavaScript/browser events a chance to run.
```

Reply:
done.